ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -410,260 +386,304 @@ private void compileInternal(String command, boolean deferClose) throws CommandP
-        compileInternal(command, true);
-      } else {
-        // Since we're reusing the compiled plan, we need to update its start time for current run
-        driverContext.getPlan().setQueryStartTime(driverContext.getQueryDisplay().getQueryStartTime());
-      }
+    context.setHiveTxnManager(driverContext.getTxnManager());
+    context.setStatsSource(driverContext.getStatsSource());
+    context.setHDFSCleanup(true);
+
+    driverTxnHandler.setContext(context);
+  }
+
+  private void setQueryId() {
+    String queryId = Strings.isNullOrEmpty(driverContext.getQueryState().getQueryId()) ?
+        QueryPlan.makeQueryId() : driverContext.getQueryState().getQueryId();
+
+    SparkSession ss = SessionState.get().getSparkSession();
+    if (ss != null) {
+      ss.onQuerySubmission(queryId);
+    }
+    driverContext.getQueryDisplay().setQueryId(queryId);
+
+    setTriggerContext(queryId);
+  }
+
+  private void setTriggerContext(String queryId) {
+    long queryStartTime;
+    // query info is created by SQLOperation which will have start time of the operation. When JDBC Statement is not
+    // used queryInfo will be null, in which case we take creation of Driver instance as query start time (which is also
+    // the time when query display object is created)
+    if (driverContext.getQueryInfo() != null) {
+      queryStartTime = driverContext.getQueryInfo().getBeginTime();
+    } else {
+      queryStartTime = driverContext.getQueryDisplay().getQueryStartTime();
+    }
+    WmContext wmContext = new WmContext(queryStartTime, queryId);
+    context.setWmContext(wmContext);
+  }
 
-      //Reset the PerfLogger so that it doesn't retain any previous values.
-      // Any value from compilation phase can be obtained through the map set in queryDisplay during compilation.
-      PerfLogger perfLogger = SessionState.getPerfLogger(true);
+  private void compileFinished(boolean deferClose) {
+    if (DriverState.getDriverState().isAborted() && !deferClose) {
+      closeInProcess(true);
+    }
+  }
 
-      // the reason that we set the txn manager for the cxt here is because each
-      // query has its own ctx object. The txn mgr is shared across the
-      // same instance of Driver, which can run multiple queries.
-      context.setHiveTxnManager(driverContext.getTxnManager());
+  /**
+   * @return The current query plan associated with this Driver, if any.
+   */
+  @Override
+  public QueryPlan getPlan() {
+    return driverContext.getPlan();
+  }
 
-      DriverUtils.checkInterrupted(driverState, driverContext, "at acquiring the lock.", null, null);
+  /**
+   * @return The current FetchTask associated with the Driver's plan, if any.
+   */
+  @Override
+  public FetchTask getFetchTask() {
+    return driverContext.getFetchTask();
+  }
 
-      lockAndRespond();
+  public void releaseLocksAndCommitOrRollback(boolean commit) throws LockException {
+    releaseLocksAndCommitOrRollback(commit, driverContext.getTxnManager());
```

Comment:
We still do not want to get rid of this method here and call the txnmanager directly?